### ISSUE TITLE
Unable to sign document after editing PDF

### ISSUE THREAD

**Issue by moritz76** [LibreSign/libresign#3949]
**Is your feature request related to a problem? Please describe.**
After editing a PDF in the browser to highlight something or insert an image, LibreSign throws an error when I try to sign the document: "Internal error. Please contact an administrator."

In the Nextcloud log, I find a lot of entries related to this action:
<details>
[libresign] Fehler: [{"file":"\/var\/www\/nextcloud\/apps\/libresign\/lib\/Handler\/JSignPdfHandler.php","line":105,"function":"signWrapper","class":"OCA\\Libresign\\Handler\\JSignPdfHandler","type":"->"},{"file":"\/var\/www\/nextcloud\/apps\/libresign\/lib\/Handler\/JSignPdfHandler.php","line":76,"function":"signUsingVisibleElements","class":"OCA\\Libresign\\Handler\\JSignPdfHandler","type":"->"},{"file":"\/var\/www\/nextcloud\/apps\/libresign\/lib\/Handler\/Pkcs12Handler.php","line":133,"function":"sign","class":"OCA\\Libresign\\Handler\\JSignPdfHandler","type":"->"},{"file":"\/var\/www\/nextcloud\/apps\/libresign\/lib\/Service\/SignFileService.php","line":275,"function":"sign","class":"OCA\\Libresign\\Handler\\Pkcs12Handler","type":"->"},{"file":"\/var\/www\/nextcloud\/apps\/libresign\/lib\/Controller\/SignFileController.php","line":129,"function":"sign","class":"OCA\\Libresign\\Service\\SignFileService","type":"->"},{"file":"\/var\/www\/nextcloud\/apps\/libresign\/lib\/Controller\/SignFileController.php","line":95,"function":"sign","class":"OCA\\Libresign\\Controller\\SignFileController","type":"->"},{"file":"\/var\/www\/nextcloud\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":208,"function":"signUsingUuid","class":"OCA\\Libresign\\Controller\\SignFileController","type":"->"},{"file":"\/var\/www\/nextcloud\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":114,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"\/var\/www\/nextcloud\/lib\/private\/AppFramework\/App.php","line":161,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"\/var\/www\/nextcloud\/lib\/private\/Route\/Router.php","line":302,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"\/var\/www\/nextcloud\/ocs\/v1.php","line":43,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"\/var\/www\/nextcloud\/ocs\/v2.php","line":7,"args":["\/var\/www\/nextcloud\/ocs\/v1.php"],"function":"require_once"}]
	POST /ocs/v2.php/apps/libresign/api/v1/sign/uuid/9e99a3a6-c111-4090-8b80-7dd41c630c5f
	von 2003:c1:9f04:6500:f329:f61d:e252:e99f von -- um 15.11.2024, 12:21:36
[libresign] Fehler: Error to sign PDF. ["FINE Default property file doesn't exists.","FINE Default property file doesn't exists.","INFO Checking input and output PDF paths.","INFO Getting key alias","INFO Used key alias: moritz busch","INFO Loading private key","INFO Getting certificate chain","INFO Opening input PDF file: \/tmp\/7b089a15366fcdc896a299330fbaf815.pdf","INFO Creating output PDF file: \/tmp\/7b089a15366fcdc896a299330fbaf815_signed.pdf","INFO Creating signature","SEVERE Problem occured","com.lowagie.text.DocumentException: Append mode requires a document without errors even if recovery was possible.","\tat com.lowagie.text.pdf.PdfStamperImp.<init>(PdfStamperImp.java:123)","\tat com.lowagie.text.pdf.PdfStamper.<init>(PdfStamper.java:130)","\tat com.lowagie.text.pdf.PdfStamper.createSignature(PdfStamper.java:707)","\tat net.sf.jsignpdf.SignerLogic.signFile(SignerLogic.java:204)","\tat net.sf.jsignpdf.Signer.signFiles(Signer.java:246)","\tat net.sf.jsignpdf.Signer.main(Signer.java:139)","","INFO Finished: Creating of signature failed."]
	POST /ocs/v2.php/apps/libresign/api/v1/sign/uuid/9e99a3a6-c111-4090-8b80-7dd41c630c5f
	von 2003:c1:9f04:6500:f329:f61d:e252:e99f von -- um 15.11.2024, 12:21:36

[libresign] Fehler: Error at JSignPdf side. LibreSign can not do nothing. Follow the error message: Error to sign PDF. ["FINE Default property file doesn't exists.","FINE Default property file doesn't exists.","INFO Checking input and output PDF paths.","INFO Getting key alias","INFO Used key alias: moritz busch","INFO Loading private key","INFO Getting certificate chain","INFO Opening input PDF file: \/tmp\/7b089a15366fcdc896a299330fbaf815.pdf","INFO Creating output PDF file: \/tmp\/7b089a15366fcdc896a299330fbaf815_signed.pdf","INFO Creating signature","SEVERE Problem occured","com.lowagie.text.DocumentException: Append mode requires a document without errors even if recovery was possible.","\tat com.lowagie.text.pdf.PdfStamperImp.<init>(PdfStamperImp.java:123)","\tat com.lowagie.text.pdf.PdfStamper.<init>(PdfStamper.java:130)","\tat com.lowagie.text.pdf.PdfStamper.createSignature(PdfStamper.java:707)","\tat net.sf.jsignpdf.SignerLogic.signFile(SignerLogic.java:204)","\tat net.sf.jsignpdf.Signer.signFiles(Signer.java:246)","\tat net.sf.jsignpdf.Signer.main(Signer.java:139)","","INFO Finished: Creating of signature failed."]
	POST /ocs/v2.php/apps/libresign/api/v1/sign/uuid/9e99a3a6-c111-4090-8b80-7dd41c630c5f
	von 2003:c1:9f04:6500:f329:f61d:e252:e99f von -- um 15.11.2024, 12:21:36
</details>

**Describe the solution you'd like**
Edit a document and then sign it.

**Comment by vitormattos** [comment]
Can you provide more details about what you mean by "editing pdf"? Perhaps step by step so I can reproduce it.

It seems like a very specific scenario and from the message, it seems that somehow the PDF has something that is being considered corrupted.

Can you share this PDF with me? You can attach it to this issue. If it is not possible to attach it to the issue because it may contain sensitive data, you can send it to me by email or Telegram. My contacts are on my GitHub profile.

**Comment by moritz76** [comment]
I have also attached the sample document

**Comment by moritz76** [comment]
Thanks for your quick reply!

By "edit pdf" I mean that when you open a PDF document in a web browser such as Firefox, you have the option to write, draw or insert an image:
![Bildschirmfoto vom 2024-11-15 16-24-07](https://github.com/user-attachments/assets/b80fc810-388c-476d-b7e2-93270b0228f0)

It does not matter which pdf.

Here is what I see when I try to sign this edited PDF:
![Bildschirmfoto vom 2024-11-15 16-26-03](https://github.com/user-attachments/assets/790277d9-a821-46b6-95bc-8d98e5a895e9)

I have also attached a sample document.
[Testdocument-1.pdf](https://github.com/user-attachments/files/17777690/Testdocument-1.pdf)

**Comment by moritz76** [comment]
Yes, the error only occurs after the document has been edited. It works fine without that. But in most cases I have to add a date or place to a document and then sign it.

Okay, this is weird. I have tested it on several Nextcloud instances and I get this error on every one of them.

Could you maybe open a document in Firefox, edit it and then try to sign it?

Thank you very much!

**Comment by vitormattos** [comment]
@moritz76 could you provide the steps that you made to do this? I used the document that you made changes and was possible sign the document.

**Comment by moritz76** [comment]
I have made a video about it:
[Bildschirmaufzeichnung vom 19.11.2024, 16:52:13.webm](https://github.com/user-attachments/assets/b3562eb6-1bdf-4a8e-a3a6-f5b2cc00ccc3)

**Comment by vitormattos** [comment]
Curious, I made the same and can't reproduce the error.

Follow the steps of my scenario:

Go to app Files
Open a pdf file at Viewer clicking on PDF at files list
Edit the PDF adding text and writing
At PDF.js editor, click on save button
Click with right button at edited file
CLick in "Open in LibreSign"
Add herself as signer
You need to see the changed PDF file
Add a visible signature
Sign the document
Need to see "Congratulations you have digitally signed a document using LibreSign"

**Comment by moritz76** [comment]
But this is basically the same thing I show in the video?

**Comment by vitormattos** [comment]
Since we made the same, maybe could be related to your environment.

Let's check if have anything else that could affect your environment:

- Screenshot of "`Configuration check`" section from `Administration settings > LibreSign`? You can replace this by the output of command: `occ libresign:configure:check` Is the same information.
- Operational System
- PHP version
- LibreSign version

Are you using SELinux?

Your error is at JSignPDF side, other possible test is reproduce the signature using the same data that LibreSign used to sign the document.

You can get the command changing this file:
libresign/vendor/jsignpdf/jsignpdf-php/src/Sign/JSignService.php
At row 26 you will found `$commandSign = $this->commandSign($params);`
After this row, add the follow:

```php
$commandSign = $this->commandSign($params);
file_put_contents('libresign.log', $commandSign);
exec($commandSign, $output);
```

Then, try again to sign the file. After this, you will found the file libresign.log at root folder of your server. If you don't find it, run this: `find / -name libresign.log` and try to run the same command that will stay inside the libresign.log file using the same user that PHP is using or also using other user. Identify the user that you will use could be important to identify what's happening because could be related about polices or anything else related to the user that PHP use to be executed by your HTTP server. Maybe will be necessary change the path of files replacing the temp files by real files.

**Comment by moritz76** [comment]
Thank you very much for your detailed reply!

- Operating System: Ubuntu 24.04
-  php version 8.3
- LibreSign Version: 10.4.1


Output of command occ libresign:configure:check
```
 success   java                Java version: openjdk version "21.0.2" 2024-01-16 LTS                                                         
  success   java                Java binary: /var/nc_data/appdata_ocqsqpspe1o3/libresign/x86_64/linux/java/jdk-21.0.2+13-jre/bin/java         
  success   pdftk               PDFtk version: 3.3.3                                                                                          
  success   pdftk               PDFtk path: /var/nc_data/appdata_ocqsqpspe1o3/libresign/x86_64/pdftk/pdftk.jar                                
  success   jsignpdf            JSignPdf version: 2.2.2                                                                                       
  success   jsignpdf            JSignPdf path: /var/nc_data/appdata_ocqsqpspe1o3/libresign/x86_64/jsignpdf/jsignpdf-2.2.2/JSignPdf.jar        
  success   openssl-configure   Root certificate setup is working fine. 
```

I add the new line in libresign/vendor/jsignpdf/jsignpdf-php/src/Sign/JSignService.php

After trying to sign an edited PDF, I can see this in the libresign.log file:
`/var/nc_data/appdata_ocqsqpspe1o3/libresign/x86_64/linux/java/jdk-21.0.2+13-jre/bin/java -jar /var/nc_data/appdata_ocqsqpspe1o3/libresign/x86_64/jsignpdf/jsignpdf-2.2.2/JSignPdf.jar /tmp/5d90a471b23c8c306df01ddc80e69fb9.pdf -ksf /tmp/5d90a471b23c8c306df01ddc80e69fb9.pfx -ksp 'f9227f748f2116d7450dfc3758873423433e5a96' -a -kst PKCS12 -pg 1 -llx 49 -lly 524 -urx 399 -ury 624 --l2-text "" -V --bg-path /tmp/oc_tmp_uf5Vq5-.png -d /tmp/`

When execute the command I get this:
```
sudo -u www-data /var/nc_data/appdata_ocqsqpspe1o3/libresign/x86_64/linux/java/jdk-21.0.2+13-jre/bin/java -jar /var/nc_data/appdata_ocqsqpspe1o3/libresign/x86_64/jsignpdf/jsignpdf-2.2.2/JSignPdf.jar /tmp/5d90a471b23c8c306df01ddc80e69fb9.pdf -ksf /tmp/5d90a471b23c8c306df01ddc80e69fb9.pfx -ksp 'f9227f748f2116d7450dfc3758873423433e5a96' -a -kst PKCS12 -pg 1 -llx 49 -lly 524 -urx 399 -ury 624 --l2-text "" -V --bg-path /tmp/oc_tmp_uf5Vq5-.png -d /tmp/
FINE Default property file doesn't exists.
FINE Default property file doesn't exists.
File /tmp/5d90a471b23c8c306df01ddc80e69fb9.pdf is not readable by the application. Check if the file exists and the user has read right.

```
Same response when running as root or www-data
Something is wrong, the file does not exist. I do not understand why the /tmp path is set here?

**Comment by vitormattos** [comment]
> I do not understand why the /tmp path is set here?

Have 3 temp files created at /tmp folder and the tmp folder and the tmp folder is get from here:

https://github.com/LibreSign/libresign/blob/c8eacbe4a9e89a0bf50676c661a837c15355499b/lib/Handler/JSignPdfHandler.php#L51

The first file is the path to file to sign. In your case you can change to the absolute path to pdf file that you wish to sign.

The second is the path to your .pfx file, this is the digital certificate that you will use to sign the document, after this, at `-ksp` is the password to your certificate file.

**Comment by vitormattos** [comment]
Manual testing before the next release is appreciated.

**Comment by vitormattos** [comment]
@asta-tud-deploy and @moritz76 
I created a new release, could you check if the PR #4127 fixed this issue?

**Comment by moritz76** [comment]
The language of the operating system is German. Thanks for the new version, but it still does not work for me.

`{"reqId":"E7Vf0ySMVKQPxWzsb161","level":3,"time":"2024-12-14T10:01:45+01:00","remoteAddr":"::ffff:80.187.66.234","user":"--","app":"libresign","method":"POST","url":"/ocs/v2.php/apps/libresign/api/v1/sign/uuid/7c7558e9-eb79-41cf-9753-ec0b9abceb9e","message":"[{\"file\":\"\\/var\\/www\\/nextcloud\\/apps\\/libresign\\/lib\\/Handler\\/JSignPdfHandler.php\",\"line\":105,\"function\":\"signWrapper\",\"class\":\"OCA\\\\Libresign\\\\Handler\\\\JSignPdfHandler\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/nextcloud\\/apps\\/libresign\\/lib\\/Handler\\/JSignPdfHandler.php\",\"line\":76,\"function\":\"signUsingVisibleElements\",\"class\":\"OCA\\\\Libresign\\\\Handler\\\\JSignPdfHandler\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/nextcloud\\/apps\\/libresign\\/lib\\/Handler\\/Pkcs12Handler.php\",\"line\":133,\"function\":\"sign\",\"class\":\"OCA\\\\Libresign\\\\Handler\\\\JSignPdfHandler\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/nextcloud\\/apps\\/libresign\\/lib\\/Service\\/SignFileService.php\",\"line\":275,\"function\":\"sign\",\"class\":\"OCA\\\\Libresign\\\\Handler\\\\Pkcs12Handler\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/nextcloud\\/apps\\/libresign\\/lib\\/Controller\\/SignFileController.php\",\"line\":129,\"function\":\"sign\",\"class\":\"OCA\\\\Libresign\\\\Service\\\\SignFileService\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/nextcloud\\/apps\\/libresign\\/lib\\/Controller\\/SignFileController.php\",\"line\":95,\"function\":\"sign\",\"class\":\"OCA\\\\Libresign\\\\Controller\\\\SignFileController\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/nextcloud\\/lib\\/private\\/AppFramework\\/Http\\/Dispatcher.php\",\"line\":208,\"function\":\"signUsingUuid\",\"class\":\"OCA\\\\Libresign\\\\Controller\\\\SignFileController\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/nextcloud\\/lib\\/private\\/AppFramework\\/Http\\/Dispatcher.php\",\"line\":114,\"function\":\"executeController\",\"class\":\"OC\\\\AppFramework\\\\Http\\\\Dispatcher\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/nextcloud\\/lib\\/private\\/AppFramework\\/App.php\",\"line\":161,\"function\":\"dispatch\",\"class\":\"OC\\\\AppFramework\\\\Http\\\\Dispatcher\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/nextcloud\\/lib\\/private\\/Route\\/Router.php\",\"line\":302,\"function\":\"main\",\"class\":\"OC\\\\AppFramework\\\\App\",\"type\":\"::\"},{\"file\":\"\\/var\\/www\\/nextcloud\\/ocs\\/v1.php\",\"line\":43,\"function\":\"match\",\"class\":\"OC\\\\Route\\\\Router\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/nextcloud\\/ocs\\/v2.php\",\"line\":7,\"args\":[\"\\/var\\/www\\/nextcloud\\/ocs\\/v1.php\"],\"function\":\"require_once\"}]","userAgent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:133.0) Gecko/20100101 Firefox/133.0","version":"30.0.4.1","data":{"app":"libresign"},"id":"675d4982538fd"}`

`[libresign] Fehler: Error to sign PDF. ["FINE Default property file doesn't exists.","FINE Default property file doesn't exists.","INFO Checking input and output PDF paths.","INFO Getting key alias","INFO Used key alias: moritz busch","INFO Loading private key","INFO Getting certificate chain","INFO Opening input PDF file: \/tmp\/51cb7fe71bba268e36d5fdfb7a9bfaff.pdf","INFO Creating output PDF file: \/tmp\/51cb7fe71bba268e36d5fdfb7a9bfaff_signed.pdf","INFO Creating signature","SEVERE Problem occured","com.lowagie.text.DocumentException: Append mode requires a document without errors even if recovery was possible.","\tat com.lowagie.text.pdf.PdfStamperImp.<init>(PdfStamperImp.java:123)","\tat com.lowagie.text.pdf.PdfStamper.<init>(PdfStamper.java:130)","\tat com.lowagie.text.pdf.PdfStamper.createSignature(PdfStamper.java:707)","\tat net.sf.jsignpdf.SignerLogic.signFile(SignerLogic.java:204)","\tat net.sf.jsignpdf.Signer.signFiles(Signer.java:246)","\tat net.sf.jsignpdf.Signer.main(Signer.java:139)","","INFO Finished: Creating of signature failed."]
	POST /ocs/v2.php/apps/libresign/api/v1/sign/uuid/7c7558e9-eb79-41cf-9753-ec0b9abceb9e
	von ::ffff:80.187.66.234 von -- um 14.12.2024, 10:01:45
`
`[libresign] Fehler: Error at JSignPdf side. LibreSign can not do nothing. Follow the error message: Error to sign PDF. ["FINE Default property file doesn't exists.","FINE Default property file doesn't exists.","INFO Checking input and output PDF paths.","INFO Getting key alias","INFO Used key alias: moritz busch","INFO Loading private key","INFO Getting certificate chain","INFO Opening input PDF file: \/tmp\/51cb7fe71bba268e36d5fdfb7a9bfaff.pdf","INFO Creating output PDF file: \/tmp\/51cb7fe71bba268e36d5fdfb7a9bfaff_signed.pdf","INFO Creating signature","SEVERE Problem occured","com.lowagie.text.DocumentException: Append mode requires a document without errors even if recovery was possible.","\tat com.lowagie.text.pdf.PdfStamperImp.<init>(PdfStamperImp.java:123)","\tat com.lowagie.text.pdf.PdfStamper.<init>(PdfStamper.java:130)","\tat com.lowagie.text.pdf.PdfStamper.createSignature(PdfStamper.java:707)","\tat net.sf.jsignpdf.SignerLogic.signFile(SignerLogic.java:204)","\tat net.sf.jsignpdf.Signer.signFiles(Signer.java:246)","\tat net.sf.jsignpdf.Signer.main(Signer.java:139)","","INFO Finished: Creating of signature failed."]
	POST /ocs/v2.php/apps/libresign/api/v1/sign/uuid/7c7558e9-eb79-41cf-9753-ec0b9abceb9e
	von ::ffff:80.187.66.234 von -- um 14.12.2024, 10:01:45
`

**Comment by vitormattos** [comment]
Hi, at newest release of LibreSing I updated the JSignPdf dependency that could affect your scenario.

Could you test again?

**Comment by moritz76** [comment]
Hi, thank you! But unfortunately the problem is still the same...

**Comment by dustbro** [comment]
I'm experiencing the same thing here. I can create a document and sign it, but if I edit the pdf after it's been created, LibreSign fails.

**Comment by vitormattos** [comment]
@dustbro sorry by delay to answer you.

Could you reproduce the same behavior until now with the latest version? If yes, could you record your screen or maybe give more details about the steps to reproduce?

**Comment by FrozenFlame** [comment]
Hey there, looks like I'm encountering this problem up to now.
[Demonstration](https://drive.google.com/file/d/1QIStEsxBnzUoLvjRSVXL-2MbqACSnjJU/view?usp=sharing)

The replication steps are followed as in video:

Replication steps
1. Annotate a PDF file with some text
2. Set up Libresign signatories
3. Attempt to sign
4. Get error

The error that showed on the Nextcloud app version of libresign:

```
The server was unable to complete your request.  
If this happens again, please send the technical details below to the server administrator.

##### Technical details:

**Remote Address**: 192.168.0.209  
**Request ID**: aK5bm6M8XceveXClwEJV  
**Message**: Error to sign PDF. ["INFO Checking input and output PDF paths.","INFO Getting key alias","INFO Used key alias: marcu","INFO Loading private key","INFO Getting certificate chain","INFO Opening input PDF file: /tmp/0ecc9fadaae67b815b4fdc873eede8c1.pdf","INFO Creating output PDF file: /tmp/0ecc9fadaae67b815b4fdc873eede8c1_signed.pdf","INFO Creating signature","SEVERE Problem occured","com.lowagie.text.DocumentException: Append mode requires a document without errors even if recovery was possible.","\tat com.lowagie.text.pdf.PdfStamperImp.<init>(PdfStamperImp.java:123)","\tat com.lowagie.text.pdf.PdfStamper.<init>(PdfStamper.java:130)","\tat com.lowagie.text.pdf.PdfStamper.createSignature(PdfStamper.java:707)","\tat net.sf.jsignpdf.SignerLogic.signFile(SignerLogic.java:204)","\tat net.sf.jsignpdf.Signer.signFiles(Signer.java:246)","\tat net.sf.jsignpdf.Signer.main(Signer.java:139)","","INFO Finished: Creating of signature failed."]
```

**Comment by vitormattos** [comment]
Nice catch @FrozenFlame, thanks for the screen recording! I’ll try to reproduce the issue.

**Comment by vitormattos** [comment]
@FrozenFlame are you using the newest release of LibreSign?
I made the same steps and worked fine to me.

**Comment by vitormattos** [comment]
@FrozenFlame could you try to change your signature engine from JSignPdf to Native?

<img width="381" height="223" alt="Image" src="https://github.com/user-attachments/assets/539fa3c4-8e84-4c36-bcf8-626426af5dca" />

I'll try with JSignPdf now, I saw now that I used Native at my test.

**Comment by vitormattos** [comment]
With JSignPdf worked fine too for me.

**Comment by FrozenFlame** [comment]
Hey there @vitormattos  yeah I'm using the latest version. Altho I see theres a new update just yesterday ;). I'll try to give JSignPdf a swing, and maybe native again after the patch.

**Comment by FrozenFlame** [comment]
Update: apparently I've always been on JSignPdf so I tried native, native failed too. Updated to the 13.2.4 update on Nextcloud and still failing for both signing engines 

The Native error is this:

```Internal Server Error

The server was unable to complete your request.
If this happens again, please send the technical details below to the server administrator.

Technical details:
Remote Address: 192.168.0.209
Request ID: ihpaXFnpgX3cYAzs9Orf
Message: Could not sign PDF using native v1 engine. Root cause: Malformed object```

The error for JSignPdf remains the same as in my video

**Comment by vitormattos** [comment]
@FrozenFlame, could you check whether the flow writes any logs to nextcloud.log? If possible, set the [log level to 0](https://docs.nextcloud.com/server/stable/admin_manual/configuration_server/logging_configuration.html) before checking the logs and then restore the previous log level after the test. Would be good do this with both signing engine. The Request ID is related to a row at your nextcloud.log file.

**Comment by FrozenFlame** [comment]
Hey there sorry just got back to this mr @vitormattos 
Yeah there aren't any apparent new error logs from what I can tell, at least not tagged as libresign
```
Error to sign PDF. ["INFO Checking input and output PDF paths.","INFO Getting key alias","INFO Used key alias: marcu","INFO Loading private key","INFO Getting certificate chain","INFO Opening input PDF file: \/tmp\/1e2c74b23da1d6684ecc4dbfe2819a4f.pdf","INFO Creating output PDF file: \/tmp\/1e2c74b23da1d6684ecc4dbfe2819a4f_signed.pdf","INFO Creating signature","SEVERE Problem occured","com.lowagie.text.DocumentException: Append mode requires a document without errors even if recovery was possible.","\tat com.lowagie.text.pdf.PdfStamperImp.<init>(PdfStamperImp.java:123)","\tat com.lowagie.text.pdf.PdfStamper.<init>(PdfStamper.java:130)","\tat com.lowagie.text.pdf.PdfStamper.createSignature(PdfStamper.java:707)","\tat net.sf.jsignpdf.SignerLogic.signFile(SignerLogic.java:204)","\tat net.sf.jsignpdf.Signer.signFiles(Signer.java:246)","\tat net.sf.jsignpdf.Signer.main(Signer.java:139)","","INFO Finished: Creating of signature failed."]
```
These are the adjacent errors to the signing error:

app: circles
message: Stopped parent traversal
```
app: no app in context:
message: dirty table reads: SELECT `filecache`.`fileid`, `storage`, `path`, `path_hash`, `filecache`.`parent`, `filecache`.`name`, `mimetype`, `mimepart`, `size`, `mtime`, `storage_mtime`, `encrypted`, `filecache`.`etag`, `filecache`.`permissions`, `checksum`, `unencrypted_size`, `metadata_etag`, `creation_time`, `upload_time`, `meta`.`json` AS `meta_json`, `meta`.`sync_token` AS `meta_sync_token` FROM `*PREFIX*filecache` `filecache` LEFT JOIN `*PREFIX*filecache_extended` `fe` ON `filecache`.`fileid` = `fe`.`fileid` LEFT JOIN `*PREFIX*files_metadata` `meta` ON `filecache`.`fileid` = `meta`.`file_id` WHERE (`path_hash` = :dcValue1) AND (`storage` = :dcValue2)
```

```
app: serverDI
message: The requested alias "OCP\IServerContainer" is deprecated. Please request "Psr\Container\ContainerInterface" directly. This alias will be removed in a future Nextcloud version.
```

Here are some other logs that seem closer related to libresign
```
str_getcsv(): the $escape parameter must be provided as its default value will change at /config/www/nextcloud/apps/libresign/lib/Handler/SignEngine/JSignPdfHandler.php#684
```

```
Exception
Error to sign PDF. ["INFO Checking input and output PDF paths.","INFO Getting key alias","INFO Used key alias: marcu","INFO Loading private key","INFO Getting certificate chain","INFO Opening input PDF file: \/tmp\/1e2c74b23da1d6684ecc4dbfe2819a4f.pdf","INFO Creating output PDF file: \/tmp\/1e2c74b23da1d6684ecc4dbfe2819a4f_signed.pdf","INFO Creating signature","SEVERE Problem occured","com.lowagie.text.DocumentException: Append mode requires a document without errors even if recovery was possible.","\tat com.lowagie.text.pdf.PdfStamperImp.<init>(PdfStamperImp.java:123)","\tat com.lowagie.text.pdf.PdfStamper.<init>(PdfStamper.java:130)","\tat com.lowagie.text.pdf.PdfStamper.createSignature(PdfStamper.java:707)","\tat net.sf.jsignpdf.SignerLogic.signFile(SignerLogic.java:204)","\tat net.sf.jsignpdf.Signer.signFiles(Signer.java:246)","\tat net.sf.jsignpdf.Signer.main(Signer.java:139)","","INFO Finished: Creating of signature failed."]
```

Sorry I can't be of much more help here.


The only adaptation I've done for now is to sign beforehand before adding the annotations or other edits

**Comment by vitormattos** [comment]
This report is from an older LibreSign/Nextcloud stack, and the signing engines have changed significantly since then. The issue was reproducible with both JSignPdf and the native engine at the time, so I don't want to assume it is fixed.

Could you please retest this with a currently supported LibreSign release and, if it still fails, attach the edited PDF if possible (or a minimal PDF that reproduces it) together with the new signing error? That would let us determine whether this is still a PDF-repair/append-mode problem or something that has already been fixed by the newer signing stack.

**Comment by FrozenFlame** [comment]
Hey there!
I'm glad to hear back from you as I tried to sign a document again some time ago, and remembered the issue when I encountered it. Funny how it works this was updated.

Anyways, I had an error here when I tried to update libresign

This app cannot be installed because the following dependencies are not fulfilled:

    Server version 34 or lower is required.

I updated my nextcloud to 35 already. Should I roll back first to 34?

**Comment by vitormattos** [comment]
No, please don't roll back Nextcloud.

LibreSign 15.x is the release line for Nextcloud 35, and v15.0.2 is currently available.

The message saying that Nextcloud 34 or lower is required means your instance is still seeing an older LibreSign release rather than the current NC35-compatible one.

Could you check which LibreSign version is currently installed and whether v15.0.2 is offered in the Apps page? If it isn't, let me know what version is shown there and we can investigate why the NC35 release is not being offered.

Once you're on LibreSign 15.x, please retry the PDF editing/signing scenario from this issue. That's the result we're interested in now.

**Comment by FrozenFlame** [comment]
Hey there, I just updated it now to the latest 15.0.3 version, and I'm getting blocked by something else, I'm on 15.0.3 now.

When I try to add a new signer to a pdf, the search is not working.
When I search the UI shows it tries to search then immediately shows an empty result.

In the network graph I find the endpoint used here /ocs/v2.php/apps/libresign/api/v1/identify-account/search?search=my_example_user&method=account

I get a 500 error with this response

{
    "ocs": {
        "meta": {
            "status": "failure",
            "statuscode": 996,
            "message": "Internal Server Error\n"
        },
        "data": []
    }
}

So I can't try to sign it right now. Trying to check my internal nextcloud logs to see what could be going on under the hood

update:
Looks like the FederatedUser is not complete issue I've been ignoring in my install is having an effect now with this app plugin now. I'll get back to this when I solve that

**Comment by vitormattos** [comment]
Thanks for the last update.

Once you're able to continue testing, please retry the original scenario from this issue with LibreSign 15.0.3:

1. Edit or annotate a PDF.
2. Add the signer.
3. Try to sign the document.

If the original PDF signing problem still happens, please report the result here.

And if the signer search issue is still happening after that, could you open a separate issue for it with the details? That will make it easier to investigate without mixing both problems.
